sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -1333,7 +1330,21 @@ case class GpuStringSplit(str: Expression, regex: Expression, limit: Expression)
   override def doColumnar(str: GpuColumnVector, regex: GpuScalar,
       limit: GpuScalar): ColumnVector = {
     val intLimit = limit.getValue.asInstanceOf[Int]
-    str.getBase.stringSplitRecord(regex.getBase, intLimit)
+    val pattern = regex.getValue.asInstanceOf[UTF8String].toString
+    if (RegexParser.isRegExpString(pattern)) {
+      val cudfPattern = new CudfRegexTranspiler(replace = false).transpile(pattern)
+      str.getBase.stringSplitRecord(
+        cudfPattern,
+        // TODO this parameter has different meaning between Java and cuDF (limit vs maxSplit)
+        intLimit,
+        true)
+    } else {
+      str.getBase.stringSplitRecord(
+        pattern,
+        // TODO this parameter has different meaning between Java and cuDF (limit vs maxSplit)
+        intLimit,
+        false)
+    }
```

Comment:
```suggestion
val splitByRegex = RegexParser.isRegExpString(pattern);
str.getBase.stringSplitRecord(pattern, intLimit, splitByRegex)
```

Reply:
Did I miss something? Why do we need to generate `cudfPattern`?

Reply:
We need to translate Java regexp to cuDF regexp to address compatibility issues